### PR TITLE
Worldinterface plugin: Implementation of loadModel; fixed makeCylinder return no name

### PR DESCRIPTION
**Pull request by torsten93**
Since I wanted to create objects in gazebo from python, I implemented the loadModel method in the worldinterface to spawn complex models.

Also the problem with the makeCylinder (#320) method is fixed by adjusted it to the makeBox method behavior.

Best,
Torsten

### CODE REVIEW

**Comment by traversaro** [comment]
Hi @torstenfol, 
   thanks for the contribution. 

A small preliminary comment: as this is a new feature, could you open the new PR against the `devel` branch? I plan to release a feature version from the devel branch before may, so I think it is a good idea to keep this  new development in devel, so that it will be part of the new 3.4 release, thanks!

**Comment by torstenfol** [comment]
Hi @traversaro,
I changed the PR base to devel branch. Is this okay or is a new PR needed?

**Comment by traversaro** [comment]
> 
> 
> Hi @traversaro,
> I changed the PR base to devel branch. Is this okay or is a new PR needed?

Thanks, this  is ok!

File: plugins/worldinterface/src/worldproxy.cpp [hunk 1]
```diff
@@ -181,7 +190,7 @@ std::string WorldProxy::makeSphere(const double radius, const GazeboYarpPlugins:
     if (collision_enable) {tmp->SetCollideMode("all");}
     else {tmp->SetCollideMode("none");}
   }
-  //HERE TMP IS ALWAYS NULL, so the following insertion is not valid. I DON'T KNOW TO FIX IT.
+  //HERE TMP IS ALWAYS NULL, so the following insertion is not valid. I DON'T KNOW TO FIX IT. -> fixed by waiting for insert
```
Comment (traversaro) [inline plugins/worldinterface/src/worldproxy.cpp:193]: Thanks, if the problem has been fixed, can you remove the old comment? Thanks!

**Comment by traversaro** [comment]
Hi @torstenfol, sorry for the late review, if you are still interested in getting this merged, I have a few observations. 

* Instead of taking an existing function and changing its signature, could you create a function with a new name (for example `loadModelFromFileWithPose` and just define the old function by calling the new function)? In this way, we avoid breaking the code of who is using the old function. 
* Thanks for sorting out the problem of the null pointer due to the missing wait. However, I think that the current logic of checking the number of models is a bit brittle: what happens if in the same time that you are trying to insert the model someone removed a model? In that case the model count changed, but it is not related to the model that you spawned. I think directly checking if the model has been spawned is less error prone. 
* To ensure that you are actually waiting for real time and not against simulated time, I would wait on the system clock, see http://www.yarp.it/classyarp_1_1os_1_1SystemClock.html#ad5fba465106ea82a1ee037c72727cced .
* It would be ideal if the timeout could be specified, but this can be done in a follow up PR, I think we can also merge this PR with the hardcoded timeout. 
* If the timeout passed and the model has not been spawned, I think we should not go forward, ma return the function with some kind of value that signal an error to the caller.

In any case, thanks a lot for the contribution! 

FYI @randaz81

**Comment by torstenfol** [comment]
Hi @traversaro thank you for the review. of course I am still interested in merging. I will include the proposed changes and update the PR.

**Comment by torsten93** [comment]
Hi @traversaro, I implemented the requested changes, including the timeout as argument for the loadModel function. The same could make sense for the make...-functions, because there is the same check.
Since this would change the signature of all this function I hardcode there for now, but if it would be okay to have it for example as an optional argument I could add this here. Otherwise it could stay for now hardcoded and it would be at any time trivial to make a new function with this as an extra argument.

**Comment by traversaro** [comment]
Thanks @torsten93 , the PR seems fine now.

**Comment by traversaro** [comment]
I would leave the PR open for a few days to see if someone else wants to review it, and then we will merge it during next week. In the meanwhile, can you add a line describing your changes in https://github.com/robotology/gazebo-yarp-plugins/blob/devel/CHANGELOG.md#unreleased ? Thanks!

**Comment by torsten93** [comment]
Okay, thanks. I will add the necessary lines in the next days.

**Comment by torsten93** [comment]
@traversaro the entries in the changelog are done

**Comment by traversaro** [comment]
Thanks @torsten93 !

**Pull request merged**